### DISCUSSION TITLE
Unable to connect to wifi after update

### DISCUSSION THREAD

**Discussion by nickthiru** [Moddable-OpenSource/moddable#1047] [Q&A] [answered]
**Development platform:**
Processor: AMD Ryzen Threadripper 3960X 24-Core Processor 3.79 GHz
System type: 64-bit operating system, x64-based processor
OS: Windows 11

**MCU:** ESP32 NodeMCU

**Description:**
I performed an update of the Moddable SDK and the ESP32 build environment as per the instructions in the "Getting Started" guide and the "Using the Moddable SDK with ESP32" guide, respectively.

When I tried to execute the code from https://github.com/Moddable-OpenSource/iot-product-dev-book/blob/master/ch3-network/wifi-code/main.js (Screenshot 1), I'm getting an error (Screenshot 2).

When I tried to execute the `http-get` example from https://github.com/Moddable-OpenSource/iot-product-dev-book/tree/master/ch3-network/http-get (Screenshot 3), nothing seems to be happening (Screenshot 4).

Note: I tried running the code on two separate nodemcu units and had the same problem.

**Screenshots:**
1:
![image](https://user-images.githubusercontent.com/47299208/221112207-f0b58579-be55-4e22-9fbe-b887d300487a.png)

2: The result in `xsbug` after running `mcconfig -d -m -p esp32/nodemcu`. In the `LOCALS` panel, you can see that `var(0)` contains the password and `var(1)` is `undefined`.

![image](https://user-images.githubusercontent.com/47299208/221113436-7af37967-7998-442a-872b-f758a34775b9.png)

3: As you can see, I didn't have to `ctrl-c` to exit the process either. Note: The last line from the `mcrun` output also has quotation marks whilst the previous ones don't.

![image](https://user-images.githubusercontent.com/47299208/221109164-3e3f227c-fb2f-4b27-8bbd-a2a7c147bd5d.png)

4: The output from `xsbug`. When I execute `mcrun`, the mod doesn't seem to run; the `xsbug` window opens but there is no output in the console. The line in the console, from the screenshot below, is from when I installed the `host` in the book chapter. As you can see, there was no output from the subsequent `mcrun` command.

![image](https://user-images.githubusercontent.com/47299208/221109249-1f86eb15-40a3-4100-8113-37a6bb19e257.png)

**References:**
1: https://github.com/Moddable-OpenSource/moddable/blob/public/documentation/Moddable%20SDK%20-%20Getting%20Started.md#win-update
2: https://github.com/Moddable-OpenSource/moddable/blob/public/documentation/devices/esp32.md#win-update

**Comment by phoddie** [comment, marked as answer]
Good catch. There was a change on ESP32 by @andycarle related to Wi-Fi and Ethernet to co-existence that broke these. A fix is pending. Thanks for the thorough report.

**Reply by nickthiru** [reply]
@phoddie Thank you for the response. I was worried that I had made a mistake on my end. I look forward to the fix.

**Reply by phoddie** [reply]
For completeness, the relevant change is [here](https://github.com/Moddable-OpenSource/moddable/commit/e3674f4a0645db6cc49032a029cf633e6ca840ba#diff-b88e3677defd5aa756778b1648cc6c945085645948390fb8fab2e279c811769b). If you locally revert that change to setup/network.js you should be good until the actual fix is available.

**Reply by nickthiru** [reply]
Noted. Thank you very much.

**Reply by nickthiru** [reply]
> For completeness, the relevant change is [here](https://github.com/Moddable-OpenSource/moddable/commit/e3674f4a0645db6cc49032a029cf633e6ca840ba#diff-b88e3677defd5aa756778b1648cc6c945085645948390fb8fab2e279c811769b). If you locally revert that change to setup/network.js you should be good until the actual fix is available.

I managed to get past installing the `host` and connecting to Wi-Fi successfully. Unfortunately, I'm still facing the same issue when I try to run `mcrun` (see Screenshots 3 & 4).

Please note that I only reverted the change in setup/network.js. Should I perhaps revert all the other files in that commit?

**Reply by andycarle** [reply]
@nickthiru The fix for this issue has been pushed as part of the [Moddable SDK 3.7.6 release](https://github.com/Moddable-OpenSource/moddable/releases/tag/3.7.6). I would recommend updating your Moddable SDK sources to that release to see if it resolves your issue.

**Reply by nickthiru** [reply]
@andycarle Thank you for the update.

**Reply by nickthiru** [reply]
@andycarle I deleted my local repo and cloned the SDK again. Unfortunately, I was still facing the same issue with `mcrun`.

I erased the device entirely and reinstalled the `host` using `mcconfig` and it was working fine as before i.e. successful Wi-Fi connection. However, when I executed `mcrun`, there was **no** output in the cmd prompt at all (previously, there was some output i.e. Screenshot 3):
![image](https://user-images.githubusercontent.com/47299208/223945982-fb983734-49eb-4442-8fca-06d3c9d97428.png)

There is not activity in `xsbug` either.

Please note that I did not follow [steps 1-9](https://github.com/Moddable-OpenSource/moddable/blob/public/documentation/devices/esp32.md#win-instructions) (for Windows) in the "Install the required drivers and development tools for the ESP32 platform" document. I had performed this update about a month ago and assumed this was not responsible for the issue I am facing.

**Reply by nickthiru** [reply]
@andycarle An update. In order to test that the issue was not with my workstation, I decided to do a brand new install and setup on my laptop.

I faced the same issue with `mcrun` as shown in Screenshots 3 and 4.

**Reply by andycarle** [reply]
Thank you for the update. I'll take a closer look at this.

**Reply by nickthiru** [reply]
@andycarle I forked and cloned the SDK. When I got to step 2 of the "Updating" section for Windows, of the "Using the Moddable SDK with ESP32" doc:

![image](https://user-images.githubusercontent.com/47299208/224604616-2638c6d7-7de9-46b0-976c-d770dea3d05f.png)

I noticed there was an error when I performed the "fetch" step:

![image](https://user-images.githubusercontent.com/47299208/224604938-45f25557-ed9e-4082-b80c-2dfb1f0229ae.png)

I'm not sure if this has some "distant" bearing on the issue with `mcrun`.

**Reply by andycarle** [reply]
Curious, but probably not directly related. If you want to tidy that up, you could always just delete the ESP-IDF and re-clone it at the right branch.

**Reply by andycarle** [reply]
@nickthiru I was able to confirm an issue with `mcrun` on Windows. I have a fix in locally. You can apply the fix yourself by making a small change to [tools/mcrun/nmake.esp32.mk](https://github.com/Moddable-OpenSource/moddable/blob/public/tools/mcrun/nmake.esp32.mk). 

Change Line 69 from:

```makefile
	set "XSBUG_PORT=$(XSBUG_PORT)" && set "XSBUG_HOST=$(XSBUG_HOST)" && $(SERIAL2XSBUG) $(UPLOAD_PORT) $(DEBUGGER_SPEED) 8N1 -install $(ARCHIVE)
```

to:

```makefile
	echo "Running serial2xsbug..." && set "XSBUG_PORT=$(XSBUG_PORT)" && set "XSBUG_HOST=$(XSBUG_HOST)" && $(SERIAL2XSBUG) $(UPLOAD_PORT) $(DEBUGGER_SPEED) 8N1 -install $(ARCHIVE)
```

And then try again (no need to even rebuild the Moddable SDK tools).

This is a little bit of voodoo, but I think the issue is that `set` has a special meaning in NMAKE and causes the rest of the line to evaluate incorrectly. Adding the echo forces everything out to the command prompt and fixes the problem.

**Reply by nickthiru** [reply]
@andycarle It's working! Explanation noted as well. Thank you very much.
